src/apps/api/utils.py

```diff
@@ -291,77 +280,40 @@ def insert(_dict, obj, pos): return {k: v for k, v in (
-
-    # Set the response type and headers to indicate a zip file download
-    buffer.seek(0)
-    response = HttpResponse(buffer, content_type='application/zip')
-    response['Content-Disposition'] = 'attachment; filename=export.zip'
+def generate_csv_response(standard, template):
+    sanitised_standard = standard.replace(' ', '_').replace('/', '-')
 
-    return response
-
-def get_standard_mapping(standard_list, template):    
-    with open(STANDARDS_MAPPING_FILE_PATH) as f:
-        standards_data = json.load(f)
-        standards_data = standards_data[0]
-
-        output_list = list()
-
-        for d in template:
-            output_dict = dict()
-            lst_dict = dict()
-            inner_dict = dict()
-            
-            copo_id = d.get('copo_id', str())       
-            
-            for key, value in d.items():
-                outer_dict = dict()
-                tol_column_dict = dict()
-                tol_column_dict[key] = dict()
-
-                for standard in standard_list:
-                    inner_dict = dict()
-
-                    if key in standards_data.keys():
-                        inner_dict[standard] = {'key': key, 'value': value} if standard == 'tol' else {'key':  standards_data.get(key, str()).get(standard, str()).get('field', str()), 'value': value}
-                    else:
-                        inner_dict[standard] = {'key': key, 'value': value} if standard == 'tol' else {'key': str(), 'value': value}
-                    
-                    # Set value to an empty string if key is unknown in the 'inner_dict'
-                    if inner_dict[standard]['key'] == '':
-                        inner_dict[standard]['value'] = str()
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = f'attachment; filename={sanitised_standard}_export.csv'
 
-                    # Merge dictionaries
-                    tol_column_dict[key] |= inner_dict
+    df = pd.DataFrame(template)
+    df.to_csv(response, index=False)
 
-                    # Merge dictionaries that reference the same TOL column name
-                    lst_dict |= tol_column_dict
+    return response
 
-            outer_dict[copo_id] = lst_dict
-            
-            # Merge dictionaries that reference the same 'copo_id'
-            output_dict |= outer_dict
+def generate_wrapper_response(error, num_found, template):
+    wrapper = get_return_template('WRAPPER')
 
-            output_list.append(output_dict)
+    if error is None:
+        if num_found == None:
+            wrapper['number_found'] = 0
+            if template == None:
+                wrapper['number_found'] = 0
+            if type(template) == type(list()):
+                wrapper['number_found'] = len(template)
+            # else:
+            #     wrapper['number_found'] = 1
+        else:
+            wrapper['number_found'] = num_found
+        wrapper['data'] = template
+        wrapper['status'] = 'OK'
```

Comment:
suggest to change it to :   wrapper['status']["error"] = False  
make it consistent in both success or error cases.